src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -1100,37 +1150,198 @@ private void PopulateParamsFromRestRequest(Dictionary<string, object?> parameter
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="context">Middleware Context.</param>
+        /// <param name="parameters">Value for the input field.</param>
+        /// <example>       1. mutation {
+        ///                 createbook(
+        ///                     item: {
+        ///                         title: "book #1",
+        ///                         reviews: [{ content: "Good book." }, { content: "Great book." }],
+        ///                         publisher: { name: "Macmillan publishers" },
+        ///                         authors: [{ birthdate: "1997-09-03", name: "Red house authors", author_name: "Dan Brown" }]
+        ///                     })
+        ///                 {
+        ///                     id
+        ///                 }
+        ///                 2. mutation {
+        ///                 createbooks(
+        ///                     items: [{
+        ///                         title: "book #1",
+        ///                         reviews: [{ content: "Good book." }, { content: "Great book." }],
+        ///                         publisher: { name: "Macmillan publishers" },
+        ///                         authors: [{ birthdate: "1997-09-03", name: "Red house authors", author_name: "Dan Brown" }]
+        ///                     },
+        ///                     {
+        ///                         title: "book #2",
+        ///                         reviews: [{ content: "Awesome book." }, { content: "Average book." }],
+        ///                         publisher: { name: "Pearson Education" },
+        ///                         authors: [{ birthdate: "1990-11-04", name: "Penguin Random House", author_name: "William Shakespeare" }]
+        ///                     }])
+        ///                 {
+        ///                     items{
+        ///                         id
+        ///                         title
+        ///                     }
+        ///                 }</example>
+        private void PopulateMutationEntityAndFieldsToAuthorize(
+            Dictionary<string, HashSet<string>> entityToFields,
+            IInputField schema,
+            string entityName,
+            IMiddlewareContext context,
+            object? parameters)
         {
-            string role = string.Empty;
-            if (context.ContextData.TryGetValue(key: AuthorizationResolver.CLIENT_ROLE_HEADER, out object? value) && value is StringValues stringVals)
+            if (parameters is List<ObjectFieldNode> listOfObjectFieldNode)
             {
-                role = stringVals.ToString();
+                // For the example createbook mutation written above, the object value for `item` is interpreted as a List<ObjectFieldNode> i.e.
+                // all the fields present for item namely- title, reviews, publisher, authors are interpreted as ObjectFieldNode.
```

Comment:
`title` is resolved as an `ObjectFieldNode` and not an `IValueNode`? title isn't an object, it's a scalar field.

Reply:
Yes, title is resolved as an `ObjectFieldNode` which represents a key-value pair where the value is a scalar value. IValueNode is the interface for `ObjectValueNode` and `ListValueNode`. For `ObjectValueNode`, the value in KV pair is an object and not a scalar value. Similarly for `ListValueNode` , the value is a list value/array value. 

Please don't confuse it with the scalar/complex field we have been referring to in the schema generation portion.